x-pack/plugins/lists/public/exceptions/hooks/persist_exception_item.test.ts

```diff
@@ -78,12 +79,45 @@ describe('usePersistExceptionItem', () => {
       >(() => usePersistExceptionItem({ http: mockKibanaHttpService, onError }));
 
       await waitForNextUpdate();
-      result.current[1](getUpdateExceptionListItemSchemaMock());
+      // NOTE: Take note here passing in an exception item where it's
+      // entries have been enriched with ids to ensure that they get stripped
+      // before the call goes through
+      result.current[1]({ ...getUpdateExceptionListItemSchemaMock(), entries: ENTRIES_WITH_IDS });
       await waitForNextUpdate();
 
       expect(result.current).toEqual([{ isLoading: false, isSaved: true }, result.current[1]]);
       expect(addExceptionItem).not.toHaveBeenCalled();
-      expect(updateExceptionItem).toHaveBeenCalled();
+      expect(updateExceptionItem).toHaveBeenCalledWith({
+        http: mockKibanaHttpService,
+        listItem: getUpdateExceptionListItemSchemaMock(),
+        signal: new AbortController().signal,
+      });
+    });
+  });
+
+  test('it invokes "addExceptionListItem" when payload has "id"', async () => {
+    const updateExceptionItem = jest.spyOn(api, 'updateExceptionListItem');
+    const createExceptionItem = jest.spyOn(api, 'addExceptionListItem');
```

Comment:
The naming here is a little confusing because it's called `addExceptionItem` in the test above this one, yet it's mocking the same thing.